Develop

sylwiavv opened the pull request:
I have one issue with ts in CompletedTaskForm.tsx on 43 line

sylwiavv committed:
Create jsno-server db, add useQuery, fetch tasks, show tasks

sylwiavv committed:
Separate task list into small parties

sylwiavv committed:
Add Dialog and fn to add a new task

sylwiavv committed:
fix some styles

sylwiavv committed:
New features

- add form to edit, delete and change progress task state
- add styles
- add hooks for setting all this thinks

sylwiavv committed:
Add state in zustand

sylwiavv committed:
Fixes

- fix bug with typescript
- fix bug with error after the other modal hasn't been closed
- iprove some styles

sylwiavv committed:
Rm bug with shadcn dialog with adding dialogDescription

sylwiavv committed:
configure json-server into vercel

sylwiavv committed:
Wip: update server.js

jugeeer commented on src/app/page.tsx:
```diff
@@ -1,21 +1,52 @@
+              setOpen={setOpen}
+              dialogTitle="Add task"
+              dialogDescription="Create a new task by providing the necessary details. Add a title, description."
+              children={<TaskAddForm setOpen={setOpen} />}
```
better to do 
`<DialogWrapper ...>
   <TaskAddForm ... />
<DialogWrapper >
`

jugeeer commented on src/app/providers.tsx:
```diff
@@ -0,0 +1,15 @@
+
+  return (
+    <QueryClientProvider client={queryClient}>
+      <FormProvider {...form}>{children}</FormProvider>
```
really bad place to initialize a FormProvider, in you case you will have one form, but this place should be use for global providers, but form it's unique

jugeeer commented on src/features/task/hooks/useAddTask.tsx:
```diff
@@ -0,0 +1,32 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
```
move all of your hooks that about work with server to folder entities,

jugeeer commented on src/features/task/hooks/useGetTasks.tsx:
```diff
@@ -0,0 +1,31 @@
+export const useTasks = () => {
+  const { setTasks } = useTasksStore();
+
+  return useQuery({
```
you file name useGetTasks for some reason you exported only one hook that name useTasks if you have onky on export in such case names should be the same

jugeeer commented on src/features/task/hooks/useGetTasks.tsx:
```diff
@@ -0,0 +1,31 @@
+  return useQuery({
+    queryKey: ['tasks'],
+    queryFn: async () => {
+      const tasks = await fetchTasks();
```
queryFn should only return a value from server and do not do any side effects

sylwiavv committed:
Update README.md

jugeeer commented on src/features/task/hooks/useSetTaskAsComplited.tsx:
```diff
@@ -0,0 +1,39 @@
+  };
+
+  return useMutation({
+    mutationFn: setTaskAsCompleted,
```
mutationFn only for server work don't do any sideEffect all sideeffect beeter to do in onSuccess

jugeeer commented on src/features/task/components/TaskList.tsx:
```diff
@@ -0,0 +1,29 @@
+  }
+
+  return (
+    <>
```
<></> why?

jugeeer commented on src/features/task/components/TaskList.tsx:
```diff
@@ -0,0 +1,29 @@
+  const { isPending, isError } = useTasks();
+  const { tasks } = useTasksStore();
+
+  if (isPending) return <p>Loading tasks...</p>;
```
good practice to use if always with {}

jugeeer commented on src/features/task/components/TaskItem.tsx:
```diff
@@ -0,0 +1,127 @@
+    if (completedDialogOpen) {
+      setCompletedDialogOpen(false);
+    }
+  }, [currentTask]);
```
very difficult useEffect to understand, you don't need useEffect in that case rewrite it

jugeeer commented on src/features/task/components/TaskItem.tsx:
```diff
@@ -0,0 +1,127 @@
+  const [deleteDialogOpen, setDeleteDialogOpen] = useState(false);
+  const [editDialogOpen, setEditDialogOpen] = useState(false);
+
+  const [completedDialogOpen, setCompletedDialogOpen] = useState(false);
```
you don't 3 states rewrite to one useState

jugeeer commented on src/features/task/components/TaskItem.tsx:
```diff
@@ -0,0 +1,127 @@
+              <Button
+                className="text-green-500 flex gap-0.15 items-center mt-4 rounded-sm"
+                onClick={() => {
+                  setCurrentTask(task);
```
you have many places where do the same setCurrentTask() and setCompletedDialogOpen (or another state), create another callback that can be used in all these cases

jugeeer commented on src/features/task/components/TaskItem.tsx:
```diff
@@ -0,0 +1,127 @@
+                />
+
+                <DialogWrapper
+                  children={<TaskEditForm />}
```
don't use children as a prop

jugeeer commented on src/features/task/components/TaskItem.tsx:
```diff
@@ -0,0 +1,127 @@
+                />
+
+                <DialogWrapper
+                  children={<CompletedTaskForm />}
```
don't use children as a prop

jugeeer commented on src/features/task/components/DeleteTaskForm.tsx:
```diff
@@ -0,0 +1,34 @@
+
+  const handleDeleteTask = () => {
+    deleteTask(id);
+  };
```
make tis callback async await for deleteTask completion and then close the modal

sylwiavv committed:
Add refactors

- move FormProvider from providers.ts
- rm tasks from zustand
- move hooks into entities folder

sylwiavv committed:
Merge branch 'develop' of github.com:sylwiavv/todo-app into develop

sylwiavv committed:
Fix type bug during adding a new task

sylwiavv committed:
Fix type scripr error

sylwiavv committed:
Add some improvements

sylwiavv committed:
Add some improvements

- rm zustand
- chcange task forms component to recive props
- changed some typos

sylwiavv committed:
Add client folder
